src/server.rs

```diff
@@ -87,15 +87,52 @@ async fn on_websocket_upgrade(socket: WebSocket) {
     };
 }
 
+fn get_index_listing(uri_path: &str) -> String {
+    let is_root = uri_path == "/";
+    let path = ROOT.get().unwrap().join(&uri_path[1..]);
+    let entries = fs::read_dir(path).unwrap();
+    let mut entry_names = entries
+        .into_iter()
+        .filter_map(|e| match e {
+            Ok(entry) => {
+                let trailing = if entry.metadata().unwrap().is_dir() {
+                    "/"
+                } else {
+                    ""
+                };
+                Some(format!(
+                    "{}{}",
+                    entry.file_name().to_str().unwrap(),
+                    trailing
+                ))
+            }
+            Err(_) => None,
+        })
+        .collect::<Vec<String>>();
+    entry_names.sort();
```

Comment:
The `..` can be added into `entries_names` here so we do not need to use another `format!()` at the end of this function.